### PR TITLE
fix: rubygems-update for github actions

### PR DESCRIPTION
**Pull request by JohnnyKei**
## Problem
GitHub Actions Ruby2.x is not working.

## Solution
set rubygems-update version

### This concerns this parts of the OpenAPI Specification:
* [EXAMPLE_LINK TO RELEVANT OPEN API SPECS PAGE](https://spec.openapis.org/oas/v3.1.0#data-types)
* [ANOTHER LINK TO RELEVANT OPEN API SPECS PAGE](https://spec.openapis.org/oas/v3.1.0#schema)

### The changes I made are compatible with:
- [ ] OAS2
- [ ] OAS3
- [ ] OAS3.1

### Related Issues
Links to any related issues.

### Checklist
- [ ] Added tests
- [ ] Changelog updated
- [ ] Added documentation to README.md
- [ ] Added example of using the enhancement into [test-app](https://github.com/rswag/rswag/tree/master/test-app)

### Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -43,7 +43,11 @@ jobs:
 
     - name: Install dependencies
       run: |
-        gem update --system
+        if [ ${{ matrix.ruby }} < 3 ]; then
```
Comment (romanblanco) [inline .github/workflows/ruby.yml]:
```suggestion
        if [ $(cut -d '.' -f 1 <<< "${{ matrix.ruby }}") -lt 3 ]; then
```

Comment (romanblanco) [reply .github/workflows/ruby.yml]: @JohnnyKei Thank you for your contributions, I'm trying to get through them now.

This seems to result in `test 2.6 < 3` and fails in Bash. The suggested change could help.

Comment (JohnnyKei) [reply .github/workflows/ruby.yml]: thanks, we try it.

Comment (JohnnyKei) [reply .github/workflows/ruby.yml]: I applied it, and it looks working.

**Review by romanblanco — APPROVED** [review]
LGTM :+1:, Thank you, for the fix, @JohnnyKei!

**Pull request merged**